Fix using non-square images with UNet2DModel and DDIM/DDPM pipelines

jenkspt opened the pull request:
Minimal changes to fix #1262.

- Corrects `UNet2DModel` docstring for parameter `sample_size`
- Adds check in DDPM/DDIM pipelines to handle tuple or int value for `sample_size` attribute of `UNet2DModel`

I still need to add tests for non square images w/ `UNet2DModel`. This is my first pull request -- I'm not to familiar with the testing yet. I'm open to suggestions on a good way to parameterize `sample_size` in `tests/test_pipelines.py`.

Also `UNet2DConditional` should also be updated ~~(as well as flax versions)~~. Not sure if this should be included in this pull request.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_1289). All of your documentation changes will be reflected on that endpoint.

patrickvonplaten commented on src/diffusers/pipelines/ddim/pipeline_ddim.py:
```diff
@@ -89,7 +89,11 @@ def __call__(
 
         # Sample gaussian noise to begin loop
-        image_shape = (batch_size, self.unet.in_channels, self.unet.sample_size, self.unet.sample_size)
+        if isinstance(self.unet.sample_size, int):
```
Isn't this the same?

patrickvonplaten reviewed (commented):
Ok for me - thanks for fixing this :-) 
Let me know when the PR is not WIP anymore.

jenkspt commented:
Added parameterized test case `test_uncond_unet_components` to `PipelineFastTests`.
In order to support this I made a few changes:
- renamed `test_components` to `test_stable_diffusion_components` to disambiguate
- removed `property` decorators from all `dummy` prefixed methods so that they can be parameterized

jenkspt commented:
Also looks like code quality checks are failing due to recent commits to main

patil-suraj reviewed (approved):
Looks good, thanks for the PR!
